### PR TITLE
feat: `avm/res/databricks/access-connector`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.databricks.access-connector](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.databricks.access-connector.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

File: avm/res/databricks/access-connector/main.bicep
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:23]:
```suggestion
@description('Optional. Enable/Disable usage telemetry for module.')
```

File: .github/CODEOWNERS
Comment (AlexanderSehr) [inline .github/CODEOWNERS:37]: Team must be created

File: avm/res/databricks/access-connector/main.bicep
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:100]:
```suggestion
output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? accessConnector.identity.?principalId : ''
```

File: .github/CODEOWNERS
Comment (AlexanderSehr) [inline .github/CODEOWNERS:37]: Team must be created

File: avm/res/databricks/access-connector/main.bicep
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:100]:
```suggestion
output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) ? accessConnector.identity.?principalId : ''
```

File: .github/CODEOWNERS
Comment (eriqua) [reply .github/CODEOWNERS:37]: And added to parent team as per https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

File: .github/workflows/avm.res.databricks.access-connector.yml
Comment (eriqua) [inline .github/workflows/avm.res.databricks.access-connector.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/CODEOWNERS
Comment (segraef) [reply .github/CODEOWNERS:37]: @Azure/avm-res-databricks-accessconnector-module-owners-bicep
@Azure/avm-res-databricks-accessconnector-module-contributors-bicep

File: avm/res/databricks/access-connector/main.bicep
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.bicep:100]:
```suggestion
output systemAssignedMIPrincipalId string = accessConnector.?identity.?principalId ?? ''
```

File: .github/CODEOWNERS
Comment (AlexanderSehr) [reply .github/CODEOWNERS:37]: Still shows errors. It might be missing the owner assignment? / it might still need approval.

Comment (segraef) [reply .github/CODEOWNERS:37]: Both teams exist and parent teams are assigned as well.

File: avm/res/databricks/access-connector/main.json
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/main.json:1]: May be outdated after recent outputs update

File: avm/res/databricks/access-connector/tests/e2e/waf-aligned/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/tests/e2e/waf-aligned/main.test.bicep:56]:
```suggestion
    location: location
```

Comment (AlexanderSehr) [reply avm/res/databricks/access-connector/tests/e2e/waf-aligned/main.test.bicep:56]: Note: Requires an update of the readme

File: avm/res/databricks/access-connector/tests/e2e/max/main.test.bicep
Comment (AlexanderSehr) [inline avm/res/databricks/access-connector/tests/e2e/max/main.test.bicep:85]:
```suggestion
    location: location
```

File: avm/res/databricks/access-connector/main.json
Comment (segraef) [reply avm/res/databricks/access-connector/main.json:1]: updated, running workflow again

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**
